js_modules/dagster-ui/packages/ui-core/src/assets/insights/AssetCatalogRateCard.tsx

```diff
@@ -9,19 +9,66 @@ export interface AssetCatalogRateCardProps {
   value: number | null; // e.g., 0.981 for 98.1%
   prevValue: number | null; // e.g., 1 for 100%
   loading: boolean;
+  unit: 'percent' | 'seconds';
+}
+
+function pct_change(x: number, y: number): number {
+  if (x === 0) {
+    if (y === 0) {
+      return 0.0; // No change
+    }
+    return Number.POSITIVE_INFINITY;
+  }
+
+  if (y > x) {
+    return Math.round(1000 * (y / x - 1)) / 10;
+  }
+  return Math.round(-(1000 * (1 - y / x))) / 10;
+}
+
+function formatValues(
+  valueOrNull: number | null,
+  prevValueOrNull: number | null,
+  unit: 'percent' | 'seconds',
+): {
+  currValueString: string;
+  prevValueString: string;
+  absDeltaString: string;
+  isDown: boolean;
+} {
+  let value = valueOrNull ?? 0;
+  let prevValue = prevValueOrNull ?? 0;
+
+  if (unit === 'percent') {
+    value = Math.round(value * 1000) / 10;
+    prevValue = Math.round(prevValue * 1000) / 10;
+  }
+
+  const deltaPct = pct_change(prevValue, value);
+  const isDown = deltaPct < 0;
+  const absDeltaPct = Math.abs(deltaPct);
+  const unitString = unit === 'percent' ? '%' : ' seconds';
+
+  return {
+    currValueString: numberFormatter.format(value) + unitString,
+    prevValueString: numberFormatter.format(prevValue) + unitString,
+    absDeltaString: numberFormatter.format(absDeltaPct) + '%', // delta is always in percent
```

Comment:
You can probably use unit formatting here, instead of concatting. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#unit_formatting